### PR TITLE
V1

### PR DESCRIPTION
**Pull request by Gamaranto**

### CODE REVIEW

File: pages/[address].tsx [hunk 1]
```diff
@@ -99,12 +100,27 @@ export const getServerSideProps: GetServerSideProps = async (ctx) => {
...
 
+    // get TVL history for this synth
+    const tvlHistory = await client.getTvl(address);
+    // compute the 24h price change
+    // const [_ydayTimestamp, ydayPrice] = await client
+    //   .getYesterdayPrice(address)
+    //   .catch((err: Error) => {
+    //     console.log(`getYesterdayPrice failed for synth: ${address}`);
+    //     return err;
+    //   });
```
Comment (mrice32) [inline pages/[address].tsx:111]: nit: comment

File: pages/[address].tsx [hunk 2]
```diff
@@ -99,12 +100,27 @@ export const getServerSideProps: GetServerSideProps = async (ctx) => {
...
+
+    const change24h = 3;
+    console.log(change24h);
```
Comment (mrice32) [inline pages/[address].tsx:114]: nit: console.log

File: pages/index.tsx [hunk 3]
```diff
@@ -34,8 +34,12 @@ export const getServerSideProps: GetServerSideProps = async () => {
   ).filter(errorFilter) as Emp[];
 
   const totalTvl = await client.getLatestTvl();
-  const totalTvm = await client.getLatestTvm();
+  // TODO: when bug in API fixed change back to all synths, for now only curated ones
```
Comment (mrice32) [inline pages/index.tsx:37]: @daywiss FYI

File: pages/index.tsx [hunk 4]
```diff
@@ -34,8 +34,12 @@ export const getServerSideProps: GetServerSideProps = async () => {
...
+  const totalTvm = await client.getLatestTvm(
+    data.map((synth) => synth.address)
+  );
 
+  console.log(totalTvm);
```
Comment (mrice32) [inline pages/index.tsx:42]: nit: console.log

File: utils/umaApi.ts [hunk 5]
```diff
@@ -94,9 +104,22 @@ const getEmpStats: GetEmpStats = async (address) => {
...
+const oneDayAgo = nDaysAgo(1);
+
+const getLatestTvl: GetStat = (address) => request("tvl", address);
+const getLatestTvm: any = (address: string[]) => request("tvm", address);
+const getTvl: any = (
+  address: string[],
+  startTimestamp = Math.floor(time90DaysAgo().toSeconds())
+) => request("tvlHistoryBetween", address, startTimestamp);
 
+const getYesterdayPrice: any = (address: string) =>
```
Comment (mrice32) [inline utils/umaApi.ts:117]: nit: do we not know the return type for this function?

**Review by mrice32 — APPROVED** [review]
LGTM!

File: pages/index.tsx (hunk 3 above)
Comment (daywiss) [reply pages/index.tsx:37]: yea will make api issue out of this

File: pages/404.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import styled from "@emotion/styled";
+import Image from "next/image";
+
+import { Link as UnstyledLink, MaxWidthWrapper } from "../components";
+//import telescope from "../public/telescope.png";
```
Comment (daywiss) [inline pages/404.tsx:6]: stray comment

File: pages/[address].tsx (hunk 1 above)
Comment (daywiss) [inline pages/[address].tsx:111]: some stray comments

File: pages/[address].tsx (hunk 2 above)
Comment (daywiss) [inline pages/[address].tsx:114]: log

File: pages/index.tsx (hunk 4 above)
Comment (daywiss) [inline pages/index.tsx:42]: can prob remove

File: utils/time.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import { DateTime } from "luxon";
+
+type NDaysAgo = (n: number) => () => DateTime;
+export const nDaysAgo: NDaysAgo = (n) => () =>
+  DateTime.now().minus({ days: n });
```
Comment (daywiss) [inline utils/time.ts:5]: might want to do start of date.now, but not a big deal

**Review by daywiss — APPROVED** [review]
looks great, thanks for working on this. small nits but not really blockers

**Pull request merged**